service/alerter/aws.go

```diff
@@ -1,10 +1,32 @@
 package alerter
 
 import (
+	"fmt"
+
 	awsresources "github.com/giantswarm/aws-operator/resources/aws"
 	"github.com/giantswarm/microerror"
 )
 
+// ListMasterInstances lists the master instances for this cluster.
+func (s *Service) ListMasterInstances(clusterID string) ([]string, error) {
+	instanceNames := []string{}
+
+	instances, err := awsresources.FindInstances(awsresources.FindInstancesInput{
+		Clients: s.awsClients,
+		Logger:  s.logger,
+		Pattern: fmt.Sprintf("%s-master-0", clusterID),
```

Comment:
Out of curiosity where does this `-0` come from? I've seen some instances with it, but AFAIK we are always using the value returned by the key method. If it is assigned by AWS when the master is duplicated, could we have `-1`, `-2`... too?

Reply:
The `-0` is because originally we launched the workers as EC2 instances not using as ASG. So the code supports numbering instances.

Masters will always be `-0` but I removed the suffix because the code does a wildcard search for instances tagged `[cluster]-master*`. I think this is a better check.

Reply:
:100: